isis/tests/FunctionalTestsCubeatt.cpp

```diff
@@ -0,0 +1,71 @@
+#include <iostream>
+
+#include "Cube.h"
+#include "Fixtures.h"
+
+#include "cubeatt.h"
+#include "gmock/gmock.h"
+
+
+using namespace Isis;
+
+static QString APP_XML = FileName("$ISISROOT/bin/xml/cubeatt.xml").expanded();
+
+// Tests setting output attributes: bit type and range
+TEST_F(SmallCube, FunctionalTestCubeattBitttypeAndRange) {
+  QString cubePath = tempDir.path() + "/bitTypeCubeatt.cub+8bit+0.0:1.0";
+
+  QVector<QString> args = {"from=" + testCube->fileName(), "to=" + cubePath};
+  UserInterface options(APP_XML, args);
+  cubeatt(options);
+
+  // Test the label
+  Cube outputCube(cubePath);
+  Pvl *label = outputCube.label();
+  PvlGroup &pixelGroup = label->findObject("IsisCube").findObject("Core").findGroup("Pixels");
+  EXPECT_EQ(QString(pixelGroup["Type"]).toStdString(), "UnsignedByte");
+  EXPECT_NEAR(pixelGroup["Base"][0].toDouble(), -0.003952569, 0.00001);
+  EXPECT_NEAR(pixelGroup["Multiplier"][0].toDouble(), 0.003952569, 0.00001);
+
+  // Test the DNs
+}
+
+TEST_F(SmallCube, FunctionalTestCubeattNoChange) {
+  QString cubePath = tempDir.path() + "/NoChangeCubeatt.cub";
+  QVector<QString> args = {"from=" + testCube->fileName() + "+1", "to=" + cubePath};
+  UserInterface options(APP_XML, args);
+  cubeatt(options);
+
+  // Test the label
+  Cube outputCube(cubePath);
+  Pvl *label = outputCube.label();
+  PvlGroup &pixelGroup = label->findObject("IsisCube").findObject("Core").findGroup("Pixels");
+  EXPECT_EQ(QString(pixelGroup["Type"]).toStdString(), "Real");
+  EXPECT_EQ(pixelGroup["Base"][0].toDouble(), 0);
+  EXPECT_EQ(pixelGroup["Multiplier"][0].toDouble(), 1);
+
+  // Test that DNs match in the input and output cubes
```

Comment:
I am confused about what this test is checking. It seems to be checking that if you don't specify any attributes that the output is the same as the input, but then it does a band selection on the input cube?

Reply:
Good point! The old test did both of these things, too, which I was confused by. I think to be actually testing what I think it needs to be testing, I'm going to take out the band selection on the input cube.